### PR TITLE
Handle session for RPs that use delegation endpoint

### PR DESCRIPTION
**Pull request by viorelaioia-zz**

### CODE REVIEW

File: pages/base.py [hunk 1]
```diff
@@ -15,6 +15,17 @@ def is_element_visible(self, *locator):
         except (NoSuchElementException, ElementNotVisibleException):
             return False
 
+    def is_element_present(self, *locator):
+        self.selenium.implicitly_wait(0)
```
Comment (johngian) [inline pages/base.py]: What does this mean? Wait 0 seconds?

File: pages/moderator.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from pages.auth0 import Auth0
+from pages.base import Base
+
+
+class Moderator(Base):
+    _login_button = (By.CSS_SELECTOR, 'a[class="login btn btn-primary"]')
+    _logout_button_locator = (By.CSS_SELECTOR, 'a[href="/oidc/logout/"]')
+
+    def __init__(self, selenium, url):
+        Base.__init__(self, selenium)
```
Comment (johngian) [inline pages/moderator.py]: super(Moderator, self).__init__(selenium)

File: pages/mozillians.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class Mozillians(Base):
+    _sign_in_button = (By.ID, 'nav-login')
+    _logout_menu_item_locator = (By.ID, 'nav-logout')
+    _profile_menu_locator = (By.CSS_SELECTOR, '#nav-main > a.dropdown-toggle')
+    _dropdown_menu_locator = (By.CSS_SELECTOR, 'ul.dropdown-menu')
+
+    def __init__(self, selenium, url):
+        Base.__init__(self, selenium)
```
Comment (johngian) [inline pages/mozillians.py]: same here

File: pages/standups.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class Standups(Base):
+    _sign_in_button_locator = (By.CSS_SELECTOR, 'a[href="/login-form/"]')
+    _sign_in_to_standup_button_locator = (By.CSS_SELECTOR, '.btn.login-button')
+    _logout_button_locator = (By.ID, 'logout-link')
+    _user_menu_locator = (By.CSS_SELECTOR, '#user-menu img')
+
+    def __init__(self, selenium, url):
+        Base.__init__(self, selenium)
```
Comment (johngian) [inline pages/standups.py]: same here.

File: tests/test_login.py [hunk 5]
```diff
@@ -3,24 +3,87 @@
...
-    def test_login(self, selenium, ldap, token, urls):
+    def test_login_testrp(self, selenium, ldap, token, urls):
         test = RpTest(selenium, urls['testRP'])
         test.login_with_ldap(ldap['email'], ldap['password'], conftest.passcode(ldap['secret_seed']))
         access_token = conftest.access_token(token)
         api = token['api']
         user_logs = requests.get(api, headers={'Authorization': 'Bearer {0}'.format(access_token)})
         initial_user_log_date = json.loads(user_logs.content)[0]['date']
-        time.sleep(16)
+        time.sleep(900)
```
Comment (johngian) [inline tests/test_login.py]: Either use a config entry for that or add a variable in the top of the file like:

```
SLEEP_TEST_LOGIN = 900
```

File: tests/test_login.py [hunk 6]
```diff
@@ -3,24 +3,87 @@
...
+
+    @pytest.mark.nondestructive
+    def test_login_mozillians(self, selenium, ldap, token, urls):
+        test = Mozillians(selenium, urls['mozillians'])
+        test.login_with_ldap(ldap['email'], ldap['password'], conftest.passcode(ldap['secret_seed']))
+        access_token = conftest.access_token(token)
+        api = token['api']
+        user_logs = requests.get(api, headers={'Authorization': 'Bearer {0}'.format(access_token)})
+        initial_user_log_date = json.loads(user_logs.content)[0]['date']
+        time.sleep(900)
```
Comment (johngian) [inline tests/test_login.py]: Same here

File: tests/test_login.py [hunk 7]
```diff
@@ -3,24 +3,87 @@
...
+
+    @pytest.mark.nondestructive
+    def test_login_moderator(self, selenium, ldap, token, urls):
+        test = Moderator(selenium, urls['moderator'])
+        test.login_with_ldap(ldap['email'], ldap['password'], conftest.passcode(ldap['secret_seed']))
+        access_token = conftest.access_token(token)
+        api = token['api']
+        user_logs = requests.get(api, headers={'Authorization': 'Bearer {0}'.format(access_token)})
+        initial_user_log_date = json.loads(user_logs.content)[0]['date']
+        time.sleep(900)
```
Comment (johngian) [inline tests/test_login.py]: Same here

File: tests/test_login.py [hunk 8]
```diff
@@ -3,24 +3,87 @@
...
+        test.logout()
+
+    def test_login_reps(self, selenium, ldap, token, urls):
+        test = Reps(selenium, urls['reps'])
+        test.login_with_ldap(ldap['email'], ldap['password'], conftest.passcode(ldap['secret_seed']))
+        access_token = conftest.access_token(token)
+        api = token['api']
+        user_logs = requests.get(api, headers={'Authorization': 'Bearer {0}'.format(access_token)})
+        initial_user_log_date = json.loads(user_logs.content)[0]['date']
+        time.sleep(900)
```
Comment (johngian) [inline tests/test_login.py]: Same here.

File: tests/test_login.py [hunk 9]
```diff
@@ -3,24 +3,87 @@
...
+        test.click_logout()
+
+    def test_login_standups(self, selenium, ldap, token, urls):
+        test = Standups(selenium, urls['standups'])
+        test.login_with_ldap(ldap['email'], ldap['password'], conftest.passcode(ldap['secret_seed']))
+        access_token = conftest.access_token(token)
+        api = token['api']
+        user_logs = requests.get(api, headers={'Authorization': 'Bearer {0}'.format(access_token)})
+        initial_user_log_date = json.loads(user_logs.content)[0]['date']
+        time.sleep(900)
```
Comment (johngian) [inline tests/test_login.py]: Same here.

**Review by johngian — APPROVED** [review]

File: pages/base.py (hunk 1 above)
Comment (viorelaioia-zz) [reply pages/base.py]: I removed that line as it is not needed

**Pull request merged**